src/v/crypto/tests/BUILD

```diff
@@ -84,3 +87,141 @@ redpanda_cc_gtest(
         "@googletest//:gtest",
     ],
 )
+
+expand_template(
+    name = "openssl_conf",
+    out = "openssl_conf.cnf",
+    substitutions = {
+        "@REDPANDA_DEPS_INSTALL_DIR@": "src/v/crypto/tests",
+    },
+    template = "test_openssl_conf.cnf.in",
+)
+
+copy_gen_file(
+    name = "fipsmodule_cnf",
+    dest = "etc/ssl/fipsmodule.cnf",
+    path = "ssl/fipsmodule.cnf",
+    source = "@openssl-fips",
+    deps = ["@openssl-fips"],
+)
+
+select_file(
+    name = "fipsmodule_module",
+    srcs = "@openssl-fips",
+    subpath = "lib/ossl-modules/fips.so",
+)
+
+redpanda_cc_bench(
+    name = "crypto_bench",
+    timeout = "short",
+    srcs = ["crypto_bench.cc"],
+    data = [
+        ":fipsmodule_cnf",
+        ":fipsmodule_module",
+        ":openssl_conf",
+    ],
+    env = {
+        "OPENSSL_CONF": "$(rootpath :openssl_conf)",
+        "MODULE_DIR": "src/v/crypto/tests/lib/ossl-modules",
+    },
+    deps = [
+        "//src/v/crypto",
+        "//src/v/random:generators",
+        "//src/v/ssx:thread_worker",
+        "@openssl-fips",
+        "@seastar",
+        "@seastar//:benchmark",
+    ],
+)
+
+redpanda_cc_bench(
+    name = "crypto_bench_fips",
+    timeout = "short",
+    srcs = ["crypto_bench.cc"],
+    data = [
+        ":fipsmodule_cnf",
+        ":fipsmodule_module",
+        ":openssl_conf",
+    ],
+    env = {
+        "OPENSSL_CONF": "$(rootpath :openssl_conf)",
+        "MODULE_DIR": "src/v/crypto/tests/lib/ossl-modules",
```

Comment:
I think we also need to capture the output of the `fipsmodule_module` in the `MODULE_DIR` in every target.
```suggestion
        "MODULE_DIR": "$(rootpath fipsmodule_module)",
```

Reply:
good catch. maybe the old path is dangling in my local build tree or something.

q: what was the failure mode you observed? was it "file not found" on module directory load, or did tests actually fail?

Reply:
as is, without the `FIPS_MODULE_REQUIRED`, the fips tests are skipped. `fips_module_present` returns `false` because it can't locate `fips.so`

https://github.com/redpanda-data/redpanda/blob/fc191eb310a77bd1e233df93f549b0744bae5a57/src/v/crypto/tests/ossl_context_service_test.cc#L65-L69

By manually disabling this check, the fips tests fail. There is no error regarding the `fips.so`. I assume that since nothing is found where `MODULE_DIR` points, the default module is loaded instead.

Reply:
I'm sort of surprised it works w/ `rootpath :fipsmodule_module`, because that _should_ be a relative path all the way to `fips.so`, where the ossl context actually wants a relative path to its parent directory. Strange. 

I guess there's some more yak shaving to do here - we might be approaching "rejigger the test" territory 🤷